### PR TITLE
Add floor ticker custom activity argument

### PR DESCRIPTION
**Pull request by rickstaa**
This PR adds the `activity` argument to the floor ticker. It follows #181.

image

### CODE REVIEW

**Comment by rssnyder** [comment]
If you could extract just the custom activity portion of this pr we could get it tested and merged while we wait for the discordgo pr.

File: go.mod [hunk 1]
```diff
@@ -13,3 +13,5 @@ require (
 	golang.org/x/text v0.3.6
 	modernc.org/sqlite v1.14.1
 )
+
+replace github.com/bwmarrin/discordgo => /home/ricks/Development/personal/discordgo
```
Comment (rssnyder) [inline go.mod:17]: remove this line

**Review by rssnyder — CHANGES_REQUESTED** [review]
You need to update all database references to account for the new attribute.

You can see all those references in this file: https://github.com/rssnyder/discord-stock-ticker/blob/master/floor_request.go

You also need to account for creating the new row when a user uses a database from a previous version that does not have this feature: https://github.com/rssnyder/discord-stock-ticker/blob/master/manager.go#L396-L420

File: go.mod (hunk 1 above)
Comment (rickstaa) [reply go.mod:17]: You are correct these are for testing the new discordgo watch feature.

**Comment by rickstaa** [comment]
@rssnyder Done!

**Comment by rickstaa** [comment]
Now let me see if I understand these databases.

**Comment by rickstaa** [comment]
@rssnyder I think I now implemented the DB code as you requested. Please take a look 👍🏻.

**Comment by rssnyder** [comment]
> @rssnyder I think I now implemented the DB code as you requested. Please take a look 👍🏻.

You had the right idea, but you only added it in one place. You still need to account for the new column in every place we deal with the db in floor_request.go.

Lines: 30, 169, 175

**Comment by rickstaa** [comment]
> 30

Good one! I overlooked the fact that we, of course, also need to save this new argument. 😅

**Comment by rickstaa** [comment]
Maybe also let's make sure the argument order is consistent in all code.

**Comment by rickstaa** [comment]
I put the `activity` after `nickname` everywhere now. I think the PR is now good to be merged.

File: floor_request.go [hunk 2]
```diff
@@ -166,13 +166,13 @@ func (m *Manager) WatchFloor(floor *Floor) {
 func (m *Manager) StoreFloor(floor *Floor) {
 
 	// store new entry in db
-	stmt, err := m.DB.Prepare("INSERT INTO floors(clientId, token, nickname, marketplace, name, frequency) values(?,?,?,?,?,?)")
+	stmt, err := m.DB.Prepare("INSERT INTO floors(clientId, token, nickname, activity, marketplace, name, frequency) values(?,?,?,?,?,?)")
```
Comment (rssnyder) [inline floor_request.go:169]:
```suggestion
	stmt, err := m.DB.Prepare("INSERT INTO floors(clientId, token, nickname, activity, marketplace, name, frequency) values(?,?,?,?,?,?,?)")
```

**Review by rssnyder — CHANGES_REQUESTED** [review]
one quick thing, then ill test and merge!

**Comment by rickstaa** [comment]
Good one. I now know what to do for the other PRs. 💪🏻  Beautifull code and package!

File: floor.go [hunk 3]
```diff
@@ -104,6 +114,24 @@ func (f *Floor) watchFloorPrice() {
...
+						f.Activity = custom_activity[itr]
+						itr++
+						itrSeed++
+					} else {
+						f.Activity = custom_activity[itr]
+						itrSeed++
+					}
+				}
+
 				err = dg.UpdateGameStatus(0, activity)
```
Comment (rssnyder) [inline floor.go:135]:
```suggestion
				err = dg.UpdateGameStatus(0, f.Activity)
```

Also need this same change on line 139 but I cant make a suggestion there so youll need to do it.

This actually uses the custom activity :)

Comment (rickstaa) [reply floor.go]: Good one!

**Review by rssnyder — APPROVED** [review]

**Pull request merged**